src/odemis/acq/move.py

```diff
@@ -2262,3 +2755,164 @@ def _getCurrentAlignerPositionLabel(self) -> int:
             return LOADING_PATH
         # None of the above -> unknown position
         return UNKNOWN
+
+class SampleStage(model.Actuator):
+    """
+    Stage wrapper component which converts the stage position to the sample stage position.
+    The sample stage coordinates system is along the sample-plane which is adjusted
+    according to the pre-tilt and other factors.
+    """
+
+    def __init__(self, name: str, role: str, stage_bare: model.Actuator , posture_manager: MicroscopePostureManager, **kwargs):
+        """
+        :param name: the name of the component (usually "Sample Stage")
+        :param role: the role of the component (usually "stage")
+        :param stage_bare: the stage component to be wrapped
+        :param posture_manager: the posture manager to be used for conversion
+        :param **kwargs: additional arguments to be passed to the parent class
+        """
+
+        self._stage_bare = stage_bare
+
+        model.Actuator.__init__(self, name, role, dependencies={"under": stage_bare},
+                                axes=copy.deepcopy(self._stage_bare.axes), **kwargs)
+
+        # update related MDs
+        self.affects = []
```

Comment:
This attribute is normally provided on all `HwComponent` and should be a VA, containing the list of component *names* (not the actual component object). You're overriding it. I don't think any code uses `.affects` in the GUI, so this works... but better rename this attribute to `_affects` or `_affected_components`.